internal/fs/fs.go

```diff
@@ -205,11 +208,26 @@ func NewFileSystem(ctx context.Context, serverCfg *ServerConfig) (fuseutil.FileS
 		metricHandle:               serverCfg.MetricHandle,
 		enableAtomicRenameObject:   serverCfg.NewConfig.EnableAtomicRenameObject,
 		globalMaxWriteBlocksSem:    semaphore.NewWeighted(serverCfg.NewConfig.Write.GlobalMaxBlocks),
+		globalMaxReadBlocksSem:     semaphore.NewWeighted(serverCfg.NewConfig.Read.GlobalMaxBlocks),
 	}
 	if serverCfg.Notifier != nil {
 		fs.notifier = serverCfg.Notifier
 	}
 
+	if serverCfg.NewConfig.Read.EnableBufferedRead {
+		var err error
+		numCPU := runtime.NumCPU()
+		totalWorkers := 3 * numCPU
+		// 10% of total workers for priority, rounded up.
+		priorityWorkers := (totalWorkers + 9) / 10
+		normalWorkers := totalWorkers - priorityWorkers
+		fs.bufferedReadWorkerPool, err = workerpool.NewStaticWorkerPool(uint32(priorityWorkers), uint32(normalWorkers))
```

Comment:
It seems, we should expose other method which hides the information about the worker. Lets do that.

Also, add a comment why 3 * NumCPU? - Generic heuristic to use 2x and 3x cpu for the IO related task which is in our case.